### PR TITLE
[68] Login component

### PR DESCRIPTION
**Pull request by aetheryna**
Created the login component, 

- Implemented Iron Session :cityscape: :muscle:
image

This is created after the user is logged in and stores them in a cookie as a session, fun fact the user details are all available here.

- Implemented a working redirect, currently redirects to "/" will change this during Dashboard development. :unicorn: 
CRIT _ Login

### CODE REVIEW

**Comment by aetheryna** [comment]
Added redirection here!

File: lib/adapters/client.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { PrismaClient } from '@prisma/client'
```
Comment (bterone) [inline lib/adapters/client.js]: I think we already have a `prismaClient.js` in our adapters 😂

File: lib/auth/session.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { ironSession } from "next-iron-session";
+
+const session = ironSession({
+  cookieName: "crit-session",
+  password: process.env.SECRET_COOKIE_PASSWORD,
```
Comment (bterone) [inline lib/auth/session.js:5]: This is a small comment, but can we have a blank field in the `.env` like below:
```md
## .env
SECRET_COOKIE_PASSWORD=
```

Only reason is that we don't have a wiki, and if we're looking at the project months from now, we might not remember this specific key should be added in the `.env` 😂

**Review by bterone — APPROVED** [review]

File: lib/adapters/client.js (hunk 1 above)
Comment (aetheryna) [reply lib/adapters/client.js]: Honestly, I think a rebase f'd up 😂

File: lib/auth/session.js (hunk 2 above)
Comment (aetheryna) [reply lib/auth/session.js:5]: 👍

File: pages/api/auth/loginUser.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import nc from 'next-connect'
+import prisma from '../../../lib/adapters/client'
```
Comment (bterone) [inline pages/api/auth/loginUser.js]: Is this linking to the proper prisma client? We renamed this to prismaClient as I recall 🤔

File: test/pages/api/auth/loginUser.test.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
+import { beforeAll, afterAll, describe, expect, jest } from "@jest/globals"
+import prisma from "../../../../lib/adapters/client"
```
Comment (bterone) [inline test/pages/api/auth/loginUser.test.js]: Same as the above comment

File: test/pages/api/auth/loginUser.test.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+import bcrypt from "bcrypt"
+import nc from "next-connect"
+
+describe("Login testing",() => {
+  jest.mock("../../../../lib/adapters/client")
```
Comment (bterone) [inline test/pages/api/auth/loginUser.test.js]: Same here

**Review by bterone — APPROVED** [review]

**Pull request merged**